### ISSUE TITLE
no-member when accessing a mangled instance variable of a super class after calling constructor of super class

### ISSUE THREAD

**Issue by zeertzjq** [PyCQA/pylint#3879]
### Steps to reproduce
Lint the following code:
```
class Test:
    def __init__(self):
        self.__test = "test"


class TestSub(Test):
    def __init__(self):
        Test.__init__(self)


print(TestSub()._Test__test)
```

### Current behavior
tmp.py:11:6: E1101: Instance of 'TestSub' has no '_Test__test' member (no-member)

### Expected behavior
No errors produced.

### pylint --version output
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Sep  5 2020, 10:50:12) 
[GCC 10.2.0]

**Comment by hippo91** [comment]
@zeertzjq thanks for the report. I can reproduce it. An interesting thing is that i also get the message:
```
W0212: Access to a protected member _Test__test of a client class (protected-access)
```

**Comment by Pierre-Sassoulas** [comment]
We also get a ``a.py:3:8: W0238: Unused private member `Test.__test` (unused-private-member)`` now, but this one I think we should keep. You're not supposed to access private member like this and checking that we don't would make the unused-private-member checker very inefficient.
